### PR TITLE
Clean up block padding rules for group block children

### LINKED ISSUE #22742: Full-width blocks inside of Group blocks no longer appear full-width.
**Describe the bug**

Full width blocks inside of full-width group blocks do not take up the full width of the container. 

**To reproduce**

1. Use a theme that does not bring its own full-width styles (e.g. The Gutenberg Starter Theme)
2. Add a Group block.
3. Add a child block, make it full-width too.
4. Observe hat the child block does not extend to the edges of the Group block.
5. Set the Group block to be wide or full, observe the same issue.  

**Expected behavior**

In previous releases, full-width children took up the full width of the parent block. This seems to no longer be the case in Version 8.2.1. 

**Screenshots**

Screen Shot 2020-05-29 at 8 21 13 AM

Screen Shot 2020-05-29 at 8 21 24 AM

Screen Shot 2020-05-29 at 8 23 12 AM

### PR DESCRIPTION
**Pull request by kjellr**
Fixes #22742. 

This PR recalibrates all the Group block children CSS so that inner blocks behave as expected at all alignments. 

Before: 

- Desktop, normal-width Groups: Full-width children appeared with too much left/right padding.
- Desktop, wide-width Groups: Full-width children appeared with too much left/right padding.
- Desktop, full-width Groups: Wide-width children appeared too wide, and full-width children appeared with too much left/right padding.
- Mobile, normal-width Groups: Full-width children appeared with too much left/right padding.
- Mobile, wide-width Groups: Full-width children appeared with too much left/right padding.
Mobile, full-width Groups: Normal and wide-width children appeared too wide, and full-width children appeared with too much left/right padding.

## Screenshots
Before (Desktop)|After (Desktop)
---|---
gutenberg test_wp-admin_post php_post=1 action=edit (3)|gutenberg test_wp-admin_post php_post=1 action=edit

Before (Mobile)|After (Mobile)
---|---
gutenberg test_wp-admin_post php_post=1 action=edit (2)|gutenberg test_wp-admin_post php_post=1 action=edit (1)

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Code looks good. This tests well for me. 

Before:

Screenshot 2020-05-29 at 16 38 42

After:

Screenshot 2020-05-29 at 16 38 51

In the vanilla styles, non-fullwide-blocks inside a full-wide group are also centered. They aren't in my own theme, which probably breaks that, but this should not be representative, and (perhaps?), up to the theme.

I would love to get some more people to test this and sanity check — perhaps @ItsJonQ? — but otherwise preapproving!

**Comment by ItsJonQ** [comment]
> I would love to get some more people to test this and sanity check 

@kjellr Looks to be working for me as well!

@jasmussen The update makes sense to me. However, I'm unsure how it would play out considering (possible) theme overrides. I suppose, that's always the decision/risk we have to make with core block styles 💪 . I trust Kjell's expertise, given his experience with themes 🙌

**Comment by jasmussen** [comment]
> However, I'm unsure how it would play out considering (possible) theme overrides. I suppose, that's always the decision/risk we have to make with core block styles 💪 . I trust Kjell's expertise, given his experience with themes 🙌

I feel this. I think we have to balance what we truly believe is best, with breaking as little as possible along the way. We should always ideally strive for the most future proof solution, even if it can get a little painful along the way.

**Comment by kjellr** [comment]
>  However, I'm unsure how it would play out considering (possible) theme overrides. I suppose, that's always the decision/risk we have to make with core block styles. 

Ideally, themes shouldn't be messing with anything that involves the default block padding (which is mostly what's being changed here). In fact, I was actually drawn over here because I saw this issue happening in my theme, began messing with the default block borders, knew it was wrong, and then took a few minutes to discover that it was a broader Gutenberg issue. 😄 

I've tested this alongside a few themes to double check what happens: 

- Twenty Twenty: This _mostly_ works, but wide blocks in full-width blocks don't appear 100% correct at middle breakpoints. That issue existed with the theme before this PR though. 
- Twenty Nineteen: This doesn't work to fix the issues there. But things looked even more broken before this patch. That theme is still waiting on [this fix](https://core.trac.wordpress.org/ticket/49843) to land, which _also_ doesn't work anymore. 

In a nutshell: block alignments are broken in most themes, regardless of what we do here. I still think this PR is the best choice we've got, because it at least fixes things in themes that don't do manually override the group block styles in the editor. We can always keep iterating (and one day, we can also provide a better way for themes to adjust these).

**Comment by kjellr** [comment]
> I would love to get some more people to test this and sanity check

Just triple checking, this should be ok to land now?

**Comment by ItsJonQ** [comment]
> Just triple checking, this should be ok to land now?

@kjellr Thank you for that context! Yes, it's 👍 from me to land

**Comment by kjellr** [comment]
Thanks, folks!
